### PR TITLE
Enable constant values as inputs to linear layers in PyTorch

### PR DESCRIPTION
**Pull request by lapid92**
Enable constant values as inputs to linear layers in PyTorch
## Checklist before requesting a review:
- [ ] I set the appropriate labels on the pull request.
- [ ] I have added/updated the release note draft (if necessary).
- [ ] I have updated the documentation to reflect my changes (if necessary).
- [ ] All function and files are well documented. 
- [ ] All function and classes have type hints.
- [ ] There is a licenses in all file.
- [ ] The function and variable names are informative. 
- [ ] I have checked for code duplications.
- [ ] I have added new unittest (if necessary).

### CODE REVIEW

File: model_compression_toolkit/core/common/graph/base_node.py [hunk 1]
```diff
@@ -42,7 +42,8 @@ def __init__(self,
                  reuse_group: str = None,
                  quantization_attr: Dict[str, Any] = None,
                  has_activation: bool = True,
-                 is_custom: bool = False
+                 is_custom: bool = False,
+                 has_positional_weights: bool = False
```
Comment (elad-c) [inline model_compression_toolkit/core/common/graph/base_node.py:46]: This is not needed. See below

File: model_compression_toolkit/core/common/graph/base_node.py [hunk 2]
```diff
@@ -96,6 +99,15 @@ def get_has_activation(self):
         """
         return self.has_activation
 
+    def get_has_positional_weights(self):
```
Comment (elad-c) [inline model_compression_toolkit/core/common/graph/base_node.py:102]: change to `has_positional_weights` which is a `@property`, that checks whether there are positional weights (i.e. check for integers in the keys of the `weights` dictionary).

File: model_compression_toolkit/core/common/statistics_correction/compute_bias_correction_of_graph.py [hunk 3]
```diff
@@ -49,11 +49,15 @@ def compute_bias_correction_of_graph(graph: Graph,
         if fw_info.is_kernel_op(n.type):
             kernel_attr = fw_info.get_kernel_op_attributes(n.type)[0]
             if n.is_weights_quantization_enabled(kernel_attr):
-                _compute_bias_correction_per_candidate_qc(n,
-                                                          kernel_attr,
-                                                          fw_info,
-                                                          graph.get_in_stats_collector(n),
-                                                          fw_impl=fw_impl)
+                if n.has_positional_weights:
+                    for candidate_qc in n.candidates_quantization_cfg:
```
Comment (elad-c) [inline model_compression_toolkit/core/common/statistics_correction/compute_bias_correction_of_graph.py:53]: Add comment

File: model_compression_toolkit/core/common/substitutions/residual_collapsing.py [hunk 4]
```diff
@@ -67,7 +67,8 @@ def substitute(self,
             return graph
 
         # Check if convolution and residual satisfy the collapsing conditions, otherwise skip substitution
-        if len(graph.get_next_nodes(first_node)) > 1 or len(graph.get_prev_nodes(second_node)) != 2:
+        if (len(graph.get_next_nodes(first_node)) > 1 or len(graph.get_prev_nodes(first_node)) < 1 or
```
Comment (elad-c) [inline model_compression_toolkit/core/common/substitutions/residual_collapsing.py:70]: isn't it clearer to write `(not blabla==1)`?

File: model_compression_toolkit/core/pytorch/back2framework/instance_builder.py [hunk 5]
```diff
@@ -33,7 +33,7 @@ def node_builder(n: BaseNode) -> Module:
 
     framework_attr = copy.copy(n.framework_attr)
     node_instance = n.type(**framework_attr)
-    node_instance.load_state_dict({k: torch.tensor(v) for k, v in n.weights.items()}, strict=False)
+    node_instance.load_state_dict({k: torch.tensor(v) for k, v in n.weights.items() if isinstance(k, str)}, strict=False)
```
Comment (elad-c) [inline model_compression_toolkit/core/pytorch/back2framework/instance_builder.py:36]: Add comment to explain why only str keys are used

**Review by elad-c — CHANGES_REQUESTED** [review]

**Review by elad-c — APPROVED** [review]
